src-cpp-structure/projects/App/Modules/SharedModels/SectionModel.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "SectionItem.h"
+
+#include <QtCore>
+
+namespace Status::Shared::Models
+{
+    class SectionModel : public QAbstractListModel
```

Comment:
We should consider using abstraction around `QAbstractListModel` to avoid boilerplate with `ModelRole` and methods overriding. The quicker we do it, the better. Ideally, the only code we should write is:
SectionItem.h
```
class SectionItem : public QObject {
    Q_OBJECT
    QML_READONLY_PROPERTY (QString, id)
    QML_READONLY_PROPERTY (SectionType, id)
    ...
    QML_READONLY_PROPERTY (Model<PendingRequestsToJoin>, pendingRequestsToJoinModel)
}
```
SectionItemModel.h
```
using SectionItemModel = Model<SectionItem>;
```
We can either use [this implementation](https://github.com/Cavewhere/lib-qt-qml-tricks/tree/727d2dd8716f6c019f92d10c797ed70d7aee1495/examples/NiceModels) or implement it on our own.

Reply:
There are more benefits on the qml side using this approach in terms of updating nested list and items in whole.

Reply:
No, my point here was that it will simplify cpp side of things a lot in terms of creating models. Instead of the current (55+123) lines of code for SectionItemModel and (133+151) for SectionItem (altogether 462 lines of code), we would end up having ~40 lines of code altogether. At the same time we will save a lot of time and possible bugs in `rowCount()`, `data()`, `roleNames()` methods, which we need to implement manually every time we create a model.